### PR TITLE
Replace whitelist term for AllowList in ios documentation

### PR DESCRIPTION
**Pull request by maririos**
Fixes: https://github.com/Azure/azure-sdk/issues/1440

### CODE REVIEW

File: docs/ios/implementation.md [hunk 1]
```diff
@@ -176,7 +176,7 @@ logger.writeLog(for: "MyClient", withLevel: AzureCoreLogLevels.Verbose, message:
 > TODO: Logging (see general guidelines)
 > * Provide abstracted logger in AzureCore
 > * Use os_logger unless over-ridden
-> * whitelistHeaders & whitelistQueryParams
+> * AllowListHeaders & allowListQueryParams
```
Comment (tjprescott) [inline docs/ios/implementation.md:179]: These are currently called `allowHeaders` and `allowQueryParams` in our iOS code base.

Comment (maririos) [reply docs/ios/implementation.md]: Awesome. updated

**Comment by maririos** [comment]
@adrianhall  what do you mean with 
> Also - this needs to flow through to the Android guidelines.

**Comment by adrianhall** [comment]
The change needs to happen in the Android guidelines as well - that's all.  @anuchandy - can you do this?

**Pull request merged**